### PR TITLE
Implements stats for lxc driver

### PR DESCRIPTION
**Pull request by ashahab-altiscale**
Implements stats and fixes stats test.
Signed-off-by: Abin Shahab ashahab@altiscale.com (github: ashahab-altiscale)

### CODE REVIEW

**Comment by dineshs-altiscale** [comment]
Can we factor out functions between libcontainer / lxc to avoid copying code?

**Comment by SvenDowideit** [comment]
@crosbymichael will this go into a 1.5.x ?

**Comment by SvenDowideit** [comment]
I have just added a note to the docs that `docker stats` only works with libcontainer - this PR will need to remove those :)

**Comment by ashahab-altiscale** [comment]
@SvenDowideit Thanks, let me know when that's merged.

**Comment by SvenDowideit** [comment]
@ashahab-altiscale merged :) I look forward to it getting taken out !

**Comment by ashahab-altiscale** [comment]
@jfrazelle Can you run you lxc build on this one?

**Comment by jfrazelle** [comment]
it is running here :) https://jenkins.dockerproject.com/job/LXR-PRs/6/console

**Comment by ashahab-altiscale** [comment]
Yup I see the same issues. Do you have an email where I can send you the logs of the builds we run? Maybe you can spot the difference? I can respond to the github email, but not sure if handles attachments.

**Comment by jfrazelle** [comment]
jess@docker.com

**Comment by ashahab-altiscale** [comment]
@jfrazelle please let me know what the lxc build says about my recent change. Thanks in advance!

**Comment by ashahab-altiscale** [comment]
@jfrazelle Were you able to get this past the lxc builds?

**Comment by jfrazelle** [comment]
running here: https://jenkins.dockerproject.com/job/LXR-PRs/8/console

**Comment by ashahab-altiscale** [comment]
Nope, this one fails again I think. Would you like to try this on a vm of yours? I found an issue with the parameters to dockerinit, and fixed them over the weekend, and have run tests multiple times on my vm. Sorry for all the nagging issues with this PR.

**Comment by ashahab-altiscale** [comment]
@jfrazelle I've sent you my logs.

**Comment by jfrazelle** [comment]
Hmmm I just don't understand why I don't get that. Hmmm I will take another crack at it tomorrow sorry.

**Comment by ashahab-altiscale** [comment]
So you also don't get these errors?

**Comment by ashahab-altiscale** [comment]
@jfrazelle Were you able to reproduce this in your vm?

**Comment by jfrazelle** [comment]
Yes :( I can't get the logs you have

**Comment by ashahab-altiscale** [comment]
@jfrazelle I removed my changes regarding dockerinit. That's my latest hunch(although I cannot reproduce the issue).

**Comment by jfrazelle** [comment]
hmmmm I dont understand :( https://jenkins.dockerproject.com/job/LXR-PRs/9/console I will try again on a vm tonight

**Comment by ashahab-altiscale** [comment]
Can I somehow access a VM where you've reproduced this issue? I can do a webex session/google hangout or ssh into your VM. I'm very curious why we can never reproduce this.

Alternatively, you can run the following test: 
Create the binary by running: sudo make binary on my branch
Then replace your docker daemon with my binary, ensure you're using lxc driver, and run:
docker run -d busybox /bin/sh -c mkdir -p '/some/path' && echo -n 'holla, i am the container'
I think this is what's failing in one of the tests, and if this fails, everything ought to fail(this binary cannot launch containers).
If you see the failure, send me the logs from the daemon(if you had started the daemon with Debug on, that'd be good).

Sorry for all the time this is taking.

**Comment by ashahab-altiscale** [comment]
@jfrazelle  Have you had a chance to run the docker run command? Thanks very much for being so persistent with this.

**Comment by jfrazelle** [comment]
do you think I need to update lxc or something?

**Comment by ashahab-altiscale** [comment]
No, the 1.0.7 would work. You do have to build it just like you did in the
Dockerfile.

**Comment by ashahab-altiscale** [comment]
@jfrazelle Any luck creating a container with this branch?

**Comment by jfrazelle** [comment]
yayyyy @ashahab-altiscale it looks good https://jenkins.dockerproject.com/job/LXR-PRs/19/console, LGTM

**Comment by jfrazelle** [comment]
ugh it needs a rebase, so sorry

**Comment by ashahab-altiscale** [comment]
@jfrazelle Can you start an lxc build for this? @GordonTheTurtle Thanks for the instructions, I think I got it right this time.

**Comment by jfrazelle** [comment]
LGTM, the latest build is in the status

**Comment by ashahab-altiscale** [comment]
@jfrazelle I still see the old errors in this: https://jenkins.dockerproject.com/job/LXR-PRs/21/console

**Comment by jfrazelle** [comment]
Oh man ya I see now I'll look into it tomorrow sorry

**Comment by jfrazelle** [comment]
The tests on lxc look good! I'm rebuilding jenkins :D

**Comment by jfrazelle** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

File: daemon/execdriver/native/driver.go [hunk 1]
```diff
@@ -291,40 +289,7 @@ func (d *driver) Clean(id string) error {
...
-
-func getEnv(key string, env []string) string {
-	for _, pair := range env {
-		parts := strings.Split(pair, "=")
-		if parts[0] == key {
-			return parts[1]
-		}
-	}
-	return ""
+	return execdriver.Stats(filepath.Join(d.root, id), d.activeContainers[id].container.Cgroups.Memory, d.machineMemory)
```
Comment (cpuguy83) [inline daemon/execdriver/native/driver.go]: Looks like this introduced a panic when calling stats on non-running containers.

**Pull request merged**